sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+      @Nullable String dsn,
+      @Nullable Level minimumBreadcrumbLevel,
+      @Nullable Level minimumEventLevel,
+      final @Nullable ITransport transport) {
+    super(name, filter, null, true, null);
+    this.dsn = dsn;
+    if (minimumBreadcrumbLevel != null) {
+      this.minimumBreadcrumbLevel = minimumBreadcrumbLevel;
+    }
+    if (minimumEventLevel != null) {
+      this.minimumEventLevel = minimumEventLevel;
+    }
+    this.transport = transport;
+  }
+
+  /**
+   * Create a Sentry Appender.
+   *
+   * @param name The name of the Appender.
+   * @param filter The filter, if any, to use.
+   * @return The SentryAppender.
+   */
+  @PluginFactory
+  public static SentryAppender createAppender(
+      @PluginAttribute("name") final String name,
+      @PluginAttribute("minimumBreadcrumbLevel") final Level minimumBreadcrumbLevel,
+      @PluginAttribute("minimumEventLevel") final Level minimumEventLevel,
+      @PluginAttribute("dsn") final String dsn,
+      @PluginElement("filter") final Filter filter) {
+
+    if (name == null) {
+      LOGGER.error("No name provided for SentryAppender");
+      return null;
+    }
+    return new SentryAppender(name, filter, dsn, minimumBreadcrumbLevel, minimumEventLevel, null);
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            options.setSentryClientName(BuildConfig.SENTRY_LOG4J2_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+            Optional.ofNullable(transport).ifPresent(options::setTransport);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  public void append(LogEvent eventObject) {
+    if (eventObject.getLevel().isMoreSpecificThan(minimumEventLevel)) {
+      Sentry.captureEvent(createEvent(eventObject));
+    }
+    if (eventObject.getLevel().isMoreSpecificThan(minimumBreadcrumbLevel)) {
+      Sentry.addBreadcrumb(createBreadcrumb(eventObject));
+    }
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Log4j2 {@link LogEvent}.
+   *
+   * @param loggingEvent the log4j2 event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull LogEvent loggingEvent) {
+    final SentryEvent event =
+        new SentryEvent(DateUtils.getDateTime(new Date(loggingEvent.getTimeMillis())));
+    final Message message = new Message();
+    message.setMessage(loggingEvent.getMessage().getFormat());
+    message.setFormatted(loggingEvent.getMessage().getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getMessage().getParameters()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = loggingEvent.getThrownProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
```

Comment:
Is there something else useful on the throwableInformation?

Reply:
does the returned throwable is already built with its cause?
cus ThrowableProxy has a getCauseProxy() method and we are not using it, but I believe this is because the returned throwable is already built with it, just to confirm.

Reply:
Yes, throwable already has a cause.

Reply:
Nothing else on `throwableInformation` that we could add to the event?

Reply:
I think everything is already there. Perhaps we can do a walk through over Zoom to see all integrations in actions and with your knowledge about Sentry maybe you'll find something missing.